### ISSUE TITLE
Allow extensions to contribute nodes to the tree

### ISSUE THREAD

**Issue by adietish** [redhat-developer/intellij-kubernetes#25]
We need to allow extensions to contribute nodes to the resource tree. The example at hand is a future tekton extension which would contribute a cathegory "Tekton Pipelines" with different sub-categories and nodes. 
Here's how this looks in vscode:
![image](https://user-images.githubusercontent.com/25126/76634242-523bfa80-6546-11ea-8e1c-cfe3aa8c6e4f.png)
![image](https://user-images.githubusercontent.com/25126/76634373-74ce1380-6546-11ea-881b-29e34af921ef.png)

**Comment by adietish** [comment]
* Documentations for tekton pipelines are here: https://openshift.github.io/pipelines-docs/docs/0.10.5/index.html

* the tekton extension for the fabric8 client is here: https://github.com/fabric8io/kubernetes-client/tree/master/extensions/tekton

**Comment by adietish** [comment]
vscode extension to display tekton pipeline nodes for an openshift cluster: https://github.com/redhat-developer/vscode-openshift-tools/blob/9eebc38b1af59acf4a9d7d32c7b7cabb1b9f9480/src/extension.ts#L212

the kubernetes extension for vscode knows 2 types of categories:
* GroupingFolder (contains ResourceFolders or nodes)
* ResourceFolder (contains nodes)

The extension declares a contributor that contributes a resource folder "ImageStreams" to the grouping folder "Workloads" if the cluster is an openshift cluster:
```
clusterExplorer.nodeSources.resourceFolder("ImageStreams", "ImageStreams", "ImageStream", "ImageStream").if(isOpenShift).at("Workloads")
```

**Comment by adietish** [comment]
example that shows how intellij-tekton is using the tekton extension for the fabric8 client to retrieve pipeline runs: https://github.com/redhat-developer/intellij-tekton/blob/master/src/main/java/com/redhat/devtools/intellij/tektoncd/tkn/PipelineRunDeserializer.java

**Comment by jeffmaury** [comment]
It's the exact opposite: because of a bug in the Fabric8 Tekton extension, we had to deserialize the JSON output by our own and not using the extension. But this is fixed in 4.9.0 so we should be good to use it now
